### ISSUE TITLE
DELETEs give a false negative ERROR

### ISSUE THREAD

**Issue by brandonlogan** [jarrodek/ChromeRestClient#542]
Whenever I do an HTTP DELETE on a resource, ARC says an ERROR happened without any headers.  However, the request went through and the server is returning back a 204 status code.  ARC seems to think these requests don't go through.  This happens anytime I do a request to any URI a DELETE method for any URI that allows DELETE.  No headers, no body.

## Expected outcome
ARC Shows me the Request Headers and status code of 204 No Content

## Actual outcome
No actual request was successful, but I would not know if because ARC gave error message of: 

> There was an error during the request
> 
> Failed to construct 'Response': Response with null body status cannot have body

## Versions
App: 7.7.26.168-stable
Chrome: 48.0.2564.109
Platform: Linux (-)

## Steps to reproduce
1. Do an HTTP DELETE request to a resource that should return a 204 No Content on success

**Comment by the2hill** [comment]
I was coming here to report the same issue. I can confirm this behavior.

**Comment by brandonlogan** [comment]
Probably helps to give you the error message that ARC gives:

There was an error during the request

Failed to construct 'Response': Response with null body status cannot have body

**Comment by brandonlogan** [comment]
Same request done with curl

$ curl -v -X DELETE http://secret.uri.com/collection-of-secrets/19186718-5676-407e-b188-070e0f963318
* Hostname was NOT found in DNS cache
*   Trying 10.1.1.1...
* Connected to secret.uri.com (10.1.1.1) port 9696 (#0)
> DELETE /collection-of-secrets/19186718-5676-407e-b188-070e0f963318 HTTP/1.1
> User-Agent: curl/7.35.0
> Host: secret.uri.com:9696
> Accept: */*
> 
< HTTP/1.1 204 No Content
* Server gunicorn/19.3.0 is not blacklisted
< Server: gunicorn/19.3.0
< Date: Tue, 19 Apr 2016 23:14:56 GMT
< Connection: keep-alive
< Content-Length: 0
< 
* Connection #0 to host secret.uri.com left intact

**Comment by jarrodek** [comment]
Thank you for issue report. 

I can't reproduce the problem though. I've created a test request where the response contains only status line (without headers and without payload). 

I hound a bug in the Response object https://bugs.chromium.org/p/chromium/issues/detail?id=524500 which is identical to this. It might be an issue with your Chrome version only because newer versions seems to work. I can't find this particular build of the Chrome to confirm it. However maybe null check before constructing the response object will help. 
I've created an issue in the transport library: https://github.com/jarrodek/socket-fetch/issues/7
